miqa/core/rest/project.py

```diff
@@ -133,8 +134,11 @@ def settings_(self, request, **kwargs):
     def import_(self, request, **kwargs):
         project: Project = self.get_object()
 
-        # tasks sent to celery must use serializable arguments
-        import_data(request.user.id, project.id)
+        try:
+            # tasks sent to celery must use serializable arguments
+            import_data(request.user.id, project.id)
+        except Exception as e:
+            raise APIException(str(e))
```

Comment:
```suggestion
        # tasks sent to celery must use serializable arguments
        import_data(request.user.id, project.id)
```
IMHO the benefit of using `APIException`s is that you can throw them when you encounter an error anywhere in your code, and DRF will intelligently convert them into a response code. In this case you would add them to `import_data`, something like this:
```
    if project.import_path.endswith('.csv'):
        import_dict = import_dataframe_to_dict(pandas.read_csv(project.import_path))
    elif project.import_path.endswith('.json'):
        import_dict = json.load(open(project.import_path))
    else:
        raise APIException(f'Invalid import file {project.import_path}.', code=status.HTTP_400_BAD_REQUEST)
```
This will raise a 400 error with an `Invalid import file ...` detail field. (the default code 500 denotes an internal server error, which doesn't seem appropriate here since the user set an invalid import_path).